ql/src/java/org/apache/hadoop/hive/ql/exec/repl/incremental/IncrementalLoadTasksBuilder.java

```diff
@@ -163,6 +163,17 @@ public IncrementalLoadTasksBuilder(String dbName, String loadPath,
       taskChainTail.addDependentTask(updateIncPendTask);
       taskChainTail = updateIncPendTask;
 
+      Database dbToLoadIn = hive.getDatabase(dbName);
+      if (dbToLoadIn != null && !ReplUtils.isTargetOfReplication(dbToLoadIn)) {
```

Comment:
When would dbToLoadIn be null?

Reply:
We have one test in TestReplicationScenarios.testTaskCreationOptimization in which we may get dbToLoadIn to be null. Rest in all cases, it would be not null.

Reply:
Check the reason why the test TestReplicationScenarios.testTaskCreationOptimization  expects  this to be null.

Reply:
Test Corrected.